server/src/server.ts

```diff
@@ -82,6 +84,29 @@ export class Server {
     return sock;
   }
 
+  private updateTemplate(template: Template): Template {
+    template.hash = crypto.createHash("sha256")
+      .update(JSON.stringify(template))
+      .digest("hex");
+
+    const cb0 = "03";                                       // push
+    const cb1 = this.toLittleEndianNumber(template.height); // height
+    const cb2 = "0".repeat(80);                             // extra nonce
+    const cb3 = "2f503253482f";                             // vote
+    template.coinbase = cb0 + cb1 + cb2 + cb3;              // coinbase
```

Comment:
After manually checking I think we are fine, but it would be sweet if there was some form of assertion here that makes sure that the coinbase data has a valid length (>= 2, <= 100), so something like:
`assert(template.coinbase.length / 2 >= 2)` (`/2` because those are hexa digits and we want the length in bytes)
and `assert(template.coinbase.length / 2 <= 100)`